go-controller/pkg/clustermanager/clustermanager.go

```diff
@@ -145,6 +146,11 @@ func (cm *ClusterManager) Start(ctx context.Context) error {
 			return err
 		}
 	}
+
+	if err := cm.statusManager.Start(cm.zoneClusterController.zoneTracker.Subscribe); err != nil {
```

Comment:
I guess that what I want to say is that is weird that we involve the `zoneClusterController`, which has nothing to do with what you are zone here, just because of the convenience that is already handling node updates or it has zone in the name.

It is also using the retry framework instead of being level driven design, so that is adding more to that pile.

I think what would make sense is to do a zone tracker that tracks node updates by itself. It can be private to the status manager package. If we need to reuse it someday, we could promote it to its own package and maybe give it a subcription interface.

Reply:
Ah I see what you mean.
I agree that zoneTracker should have its own handler, which I will add. But I still would like to have it as a separate package inside status_manager, in case we need to move it somewhere later, and also it is a logically separate controller that I believe makes sense to have in an isolated package